app/services/proxy_load_balancer.rb

```diff
@@ -0,0 +1,28 @@
+class ProxyLoadBalancer
+  def initialize
+    @tasks = {}
+  end
+
+  # Get the least loaded proxy from the given list of proxies
+  def next(proxies)
+    @tasks.merge!(get_counts(proxies - @tasks.keys))
+    next_proxy = @tasks.select { |proxy, _| proxies.include?(proxy) }.min_by { |_, job_count| job_count }.first
+    @tasks[next_proxy] += 1
+    next_proxy
+  end
+
+  private
+
+  def get_counts(proxies)
+    proxies.inject({}) do |result, proxy|
+      begin
+        proxy_api = ProxyAPI::ForemanDynflow::DynflowProxy.new(:url => proxy.url)
+        result[proxy] = proxy_api.tasks_count('running')
+        result
+      rescue
+        Rails.logger.error("Could not fetch task counts from #{proxy}, skipped.")
```

Comment:
if count request fails, the caching would not work fully: the proxy that is down would get requested every time it's part of some proxy pool: we might want to store something into `result[proxy]` also, when the request fails (`nil` might be ok, if we count on that on other places